sdk/keyvault/Azure.Security.KeyVault.Certificates/src/DeleteCertificateOperation.cs

```diff
@@ -0,0 +1,145 @@
+
+                try
+                {
+                    _response = _pipeline.GetResponse(RequestMethod.Get, cancellationToken, CertificateClient.DeletedCertificatesPath, _value.Name);
+                    _completed = CheckCompleted(_response);
+                }
+                catch (Exception e)
+                {
+                    scope.Failed(e);
+                    throw;
+                }
+            }
+
+            return GetRawResponse();
+        }
+
+        /// <inheritdoc/>
+        public override async ValueTask<Response> UpdateStatusAsync(CancellationToken cancellationToken = default)
+        {
+            if (!_completed)
+            {
+                using DiagnosticScope scope = _pipeline.CreateScope("Azure.Security.KeyVault.Certificates.DeleteCertificateOperation.UpdateStatus");
+                scope.AddAttribute("secret", _value.Name);
+                scope.Start();
+
+                try
+                {
+                    _response = await _pipeline.GetResponseAsync(RequestMethod.Get, cancellationToken, CertificateClient.DeletedCertificatesPath, _value.Name).ConfigureAwait(false);
+                    _completed = await CheckCompletedAsync(_response).ConfigureAwait(false);
+                }
+                catch (Exception e)
+                {
+                    scope.Failed(e);
+                    throw;
+                }
+            }
+
+            return GetRawResponse();
+        }
+
+        /// <inheritdoc />
+        public override ValueTask<Response<DeletedCertificate>> WaitForCompletionAsync(CancellationToken cancellationToken = default) =>
+            this.DefaultWaitForCompletionAsync(s_defaultPollingInterval, cancellationToken);
+
+        /// <inheritdoc />
+        public override ValueTask<Response<DeletedCertificate>> WaitForCompletionAsync(TimeSpan pollingInterval, CancellationToken cancellationToken) =>
+            this.DefaultWaitForCompletionAsync(pollingInterval, cancellationToken);
+
+        private static async Task<bool> CheckCompletedAsync(Response response)
+        {
+            switch (response.Status)
+            {
+                case 200:
+                case 403: // Access denied but proof the certificate was deleted.
+                    return true;
+
+                case 404:
+                    return false;
+
+                default:
+                    throw await response.CreateRequestFailedExceptionAsync().ConfigureAwait(false);
+            }
+        }
+        private static bool CheckCompleted(Response response)
+        {
+            switch (response.Status)
+            {
+                case 200:
+                case 403: // Access denied but proof the certificate was deleted.
+                    return true;
+
+                case 404:
+                    return false;
+
+                default:
+                    throw response.CreateRequestFailedException();
+            }
+        }
+    }
+}
```

Comment:
Could possibly simplify this by adding one Async method with an isAsync parameter.

Reply:
One returns a task. I suppose I could change that to a `ValueTask<bool>` but this is so small I'm not sure even that overhead is worth it.